src/Compilers/CSharp/Portable/Symbols/Source/SourceMemberContainerSymbol_ImplementationChecks.cs

```diff
@@ -540,6 +535,15 @@ private void CheckMembersAgainstBaseType(
             {
                 cancellationToken.ThrowIfCancellationRequested();
 
+                // If we have invalid record inheritance (a record deriving from a non-record), skip checking the
+                // synthesized members for proper new/override usage as we're very likely to just report cascaded
```

Comment:
Is it just "very likely"? I thought it better be guaranteed otherwise we could have an invalid program for which we don't report any errors.